tests/series/test_series.py

```diff
@@ -1980,16 +1981,92 @@ def test_dtype_type() -> None:
 
 def test_types_to_numpy() -> None:
     s = pd.Series(["a", "b", "c"], dtype=str)
-    check(assert_type(s.to_numpy(), np_1darray), np_1darray)
-    check(assert_type(s.to_numpy(dtype="str", copy=True), np_1darray), np_1darray)
-    check(assert_type(s.to_numpy(na_value=0), np_1darray), np_1darray)
-    check(assert_type(s.to_numpy(na_value=np.int32(4)), np_1darray), np_1darray)
-    check(assert_type(s.to_numpy(na_value=np.float16(4)), np_1darray), np_1darray)
-    check(assert_type(s.to_numpy(na_value=np.complex128(4, 7)), np_1darray), np_1darray)
+    check(assert_type(s.to_numpy(), np_1darray[np.str_]), np_1darray)
+    check(
+        assert_type(s.to_numpy(dtype="str", copy=True), np_1darray[np.str_]), np_1darray
+    )
+    check(assert_type(s.to_numpy(na_value=0), np_1darray[np.str_]), np_1darray)
+    check(
+        assert_type(s.to_numpy(na_value=np.int32(4)), np_1darray[np.str_]), np_1darray
+    )
+    check(
+        assert_type(s.to_numpy(na_value=np.float16(4)), np_1darray[np.str_]), np_1darray
+    )
+    check(
+        assert_type(s.to_numpy(na_value=np.complex128(4, 7)), np_1darray[np.str_]),
+        np_1darray,
+    )
 
     check(assert_type(pd.Series().to_numpy(), np_1darray), np_1darray)
 
 
+def test_to_numpy() -> None:
+    """Test Series.to_numpy for different types."""
+    s1 = pd.Series(["a", "b", "c"], dtype=str)
+    check(assert_type(s1.to_numpy(), np_1darray[np.str_]), np_1darray, str)
+
+    s2 = pd.Series(["a", "b", "c"]).astype(bytes)
+    check(assert_type(s2.to_numpy(), np_1darray[np.bytes_]), np_1darray, np.bytes_)
+
+    s3 = pd.Series([True, False])
+    check(assert_type(s3.to_numpy(), np_1darray[np.bool_]), np_1darray, np.bool_)
+
+    s4 = pd.Series([2, 3, 4])
+    check(assert_type(s4.to_numpy(), np_1darray[np.integer]), np_1darray, np.integer)
+
+    s5 = pd.Series([2.0, 3.54, 4.84])
+    check(assert_type(s5.to_numpy(), np_1darray[np.floating]), np_1darray, np.floating)
+
+    s6 = pd.Series([2.0 + 2j, 3.54 + 4j, 4.84])
+    check(
+        assert_type(s6.to_numpy(), np_1darray[np.complexfloating]),
+        np_1darray,
+        np.complexfloating,
+    )
+
+    dates = pd.Series(
+        [
+            pd.Timestamp("2020-01-01"),
+            pd.Timestamp("2020-01-15"),
+            pd.Timestamp("2020-02-01"),
+        ],
+        dtype="datetime64[ns]",
+    )
+    s7 = pd.Series(pd.PeriodIndex(dates, freq="M"))
```

Comment:
There is a conversion function
```suggestion
    s7 = pd.PeriodIndex(dates, freq="M").to_series()
```